src/altair/block_processing.rs

```diff
@@ -55,9 +175,73 @@ pub fn process_deposit<
         MAX_VALIDATORS_PER_COMMITTEE,
         SYNC_COMMITTEE_SIZE,
     >,
-    _deposit: &mut Deposit,
-    _context: &Context,
+    deposit: &mut Deposit,
+    context: &Context,
 ) -> Result<()> {
+    let branch = deposit
+        .proof
+        .iter()
+        .map(|node| Node::from_bytes(node.as_ref().try_into().unwrap()))
+        .collect::<Vec<_>>();
+    let leaf = deposit.data.hash_tree_root()?;
+    let depth = DEPOSIT_CONTRACT_TREE_DEPTH + 1;
+    let index = state.eth1_deposit_index as usize;
+    let root = &state.eth1_data.deposit_root;
+    if !is_valid_merkle_branch(&leaf, branch.iter(), depth, index, root) {
+        return Err(invalid_operation_error(InvalidOperation::Deposit(
+            InvalidDeposit::InvalidProof {
+                leaf,
+                branch,
+                depth,
+                index,
+                root: *root,
+            },
+        )));
+    }
+
+    // NOTE: deviate from the order of the spec to avoid mutations
+    // that would need to be rolled back upon failure
+    let public_key = deposit.data.public_key.clone();
```

Comment:
```suggestion
    let public_key = &deposit.data.public_key;
```

it looks like this is fine and saves a clone :)